docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+However, `SetProperty<T>(ref T, T, string)` provides the same functionality with re-ordered parameters.
+
+```csharp
+// MvvmLight
+this.Set(nameof(this.MyProperty), ref this.myProperty, value);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Set<T> ( ref T, T, string )
+
+`Set<T>(ref T, T, string)` has a renamed direct replacement, `SetProperty<T>(ref T, T, string)`.
+
+```csharp
+// MvvmLight
+this.Set(ref this.myProperty, value, nameof(this.MyProperty));
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+### ObservableObject Properties
+
+#### PropertyChangedHandler
+
+`PropertyChangedHandler` does not have a direct replacement.
+
+To raise a property changed event via the `PropertyChanged` event handler, you need to call the `OnPropertyChanged` method instead. 
+
+```csharp
+// MvvmLight
+PropertyChangedEventHandler handler = this.PropertyChangedHandler;
+
+// Toolkit.Mvvm
+this.OnPropertyChanged(nameof(this.MyProperty));
+```
+
+## Migrating ViewModelBase
+
+The following steps focus on migrating your existing components which take advantage of the `ViewModelBase` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides an [`ObservableRecipient`](ObservableRecipient) type that provides similar functionality. 
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+### ViewModelBase Methods
+
+#### ICleanup.Cleanup ()
+
+There is no direct replacement for the `ICleanup` interface.
+
+However, the `ObservableRecipient` provides an `OnDeactivated` method which should be used to provide the same functionality as `Cleanup`. 
+
+```csharp
+// MvvmLight
+this.Cleanup();
+
+// Toolkit.Mvvm
+this.OnDeactivated();
+```
+
+Note, the `OnActivated` and `OnDeactivated` methods can be called from your existing solution as with `Cleanup`. 
+
+However, the `ObservableRecipient` exposes an `IsActive` property that also controls the call to these methods when it is set. 
+
+#### Set<T> ( string, ref T, T, bool )
+
+`Set<T>(string, ref T, T, bool)` does not have a like-for-like method signature replacement. 
+
+However, `SetProperty<T>(ref T, T, bool, string)` provides the same functionality with re-ordered parameters.
+
+```csharp
+// MvvmLight
+this.Set(nameof(this.MyProperty), ref this.myProperty, value, true);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, true, nameof(this.MyProperty));
+```
+
+Note, the value and broadcast boolean parameters are not optional in the Toolkit's implementation and must be provided to use this method.
```

Comment:
Should we mention that the reason the `bool broadcast` parameter is not optional is because if you remove that you just end up with the same exact signature as the equivalent `SetProperty` method in the base `ObservableObject` class? That is, if users are not using that parameter they can just call the overload from the base class and achieve the same exact behavior.
In that sense, it's basically just like if the parameter was optional 😄

Reply:
I've popped that in there too. Makes sense to call that out so folks don't think it's not achievable.